controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -2820,16 +2820,30 @@ bool master_thread::handle_cmdu_control_message(const std::string &src_mac,
             }
 
             if (!database.set_node_name(
-                    client_mac, std::string(notification->name(message::NODE_NAME_LENGTH)))) {
+                    client_mac, std::string(notification_in->name(message::NODE_NAME_LENGTH)))) {
                 LOG(ERROR) << "set node name failed";
             }
 
-            if (database.get_node_ipv4(client_mac) != ipv4) {
-                LOG(DEBUG) << "handle_completed_connection client_mac = " << client_mac;
-                son_actions::handle_completed_connection(database, cmdu_tx, tasks, client_mac);
-            }
+            if (database.is_node_wireless(client_mac)) {
+                auto notification_out = message_com::create_vs_message<
+                    beerocks_message::cACTION_CONTROL_CLIENT_NEW_IP_ADDRESS_NOTIFICATION>(cmdu_tx);
+
+                if (!notification_out) {
+                    LOG(ERROR) << "Failed building message!";
+                    return false;
+                }
+                notification_out->mac()  = notification_in->mac();
+                notification_out->ipv4() = notification_in->ipv4();
 
-            if (!database.is_node_wireless(client_mac)) {
+                auto clients_bssid_radio_mac = database.get_node_parent(client_mac);
+                if (clients_bssid_radio_mac.empty()) {
+                    LOG(WARNING) << "Client does not have a valid parent hostap on the database";
+                    return true;
+                }
+                son_actions::send_cmdu_to_agent(src_mac, cmdu_tx, database,
```

Comment:
I am clearly missing something since this is weird... The master receives a DHCP notification from the agent, and it sends back exactly the same info to the agent...

Reply:
I agree it seems weird, but we have to do it like this since the DHCP event is getting to the son_slave which controls the backhaul manager which is not necessarily the son_slave the client is connected to. The controller knows to which son_slave the client is connected to so it can start the client monitoring on the right monitor thread.
Besides that, it is more convenient that this functionality is centralized on the controller from DEBUG POV.